src/Json/JsonFile.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Copyright © 2017-2018 Maks Rafalko
+ *
+ * License: https://opensource.org/licenses/BSD-3-Clause New BSD License
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Json;
+
+use Infection\Json\Exception\JsonValidationException;
+use Infection\Json\Exception\ParseException;
+use JsonSchema\Validator;
+
+/**
+ * @internal
+ */
+final class JsonFile
+{
+    private const SCHEMA_FILE = 'file://' . __DIR__ . '/../../resources/schema.json';
+
+    /**
+     * @var string
+     */
+    private $path;
+
+    /**
+     * @var \stdClass
+     */
+    private $data;
+
+    /**
+     * @param string $path
+     */
+    public function __construct(string $path)
+    {
+        $this->path = $path;
+    }
+
+    private function validateSchema(): void
```

Comment:
Suggest to move all private methods to the bottom of the file

Reply:
@borNfreee 

There's actually the `ordered_class_elements ` fixer that does that, but it will of course also apply all of these changes to existing classes.

Reply:
I did in this PR but yeah we may enable `ordered_class_elements` in the separate PR like @localheinz suggests

Reply:
@borNfreee @sidz 

It's already enabled, actually, see:

https://github.com/infection/infection/blob/6b800242c88187b72e7bd6690071933ddb9e0877/.php_cs.dist#L58-L64

We could further configure it, though!